src/sentry/search/events/fields.py

```diff
@@ -3113,6 +3114,7 @@ def _resolve_user_display_alias(self, _: str) -> SelectType:
         columns = ["user.email", "user.username", "user.ip"]
         return Function("coalesce", [self.column(column) for column in columns], USER_DISPLAY_ALIAS)
 
+    @lru_cache
```

Comment:
nit: Do we specifically need `lru_cache`? I'm wondering if we should use `cached_property` from `django.utils.functional` to be consistent with how I did this with `project_slugs`?

Reply:
No specific reason to use `lru_cache`, we can use `cached_property` though we'd need to do a little restructuring since `cached_property` assumes the method has no arguments whereas this has 1 argument that is ignored.

Reply:
Also one question, since I'm not sure, the lru_cache is just applicable for this instance of QueryFields right? Since the key for the cache is a hash of the args/kwargs and this method doesn't really take in any args it can't be any kind of a global cache.

Reply:
Yes, the `lru_cache` is applicable to just this instance of `QueryFields`. I also switched to `cached_property` like @wmak suggested to ensure this behaviour.